apps/mobile/src/features/token/account-list-item.tsx

```diff
@@ -0,0 +1,85 @@
+import { Balance } from '@/components/balance/balance';
+import { AccountListItem as AccountListItemComponent } from '@/features/account/account-list/account-list-item';
+import { Account } from '@/store/accounts/accounts';
+import { useAccounts } from '@/store/accounts/accounts.read';
+import { WalletStore } from '@/store/wallets/utils';
+import { WalletLoader } from '@/store/wallets/wallets.read';
+import { t } from '@lingui/macro';
+
+import { ChevronRightIcon, Text } from '@leather.io/ui/native';
+
+import { AccountAvatar } from '../account/components/account-avatar';
+import { TokenDetailsCard } from './components/token-details-card';
+import { useGetAccountTokenBalance } from './hooks/use-get-token-balance';
+
+export function AccountList({
+  tokenId,
+  selectAccount,
+}: {
+  tokenId: string;
+  selectAccount: (account: Account) => void;
+}) {
+  const accounts = useAccounts();
+  return (
+    <TokenDetailsCard title={t({ id: 'token.details.accounts_title', message: 'Accounts' })}>
+      {accounts.list
+        .filter(account => account.status !== 'hidden')
+        .map(account => (
+          <WalletLoader fingerprint={account.fingerprint} key={account.id}>
+            {wallet => (
+              <AccountListItem
+                key={account.id}
+                account={account}
+                wallet={wallet}
+                tokenId={tokenId}
+                onPress={() => selectAccount(account)}
+              />
+            )}
+          </WalletLoader>
+        ))}
+    </TokenDetailsCard>
+  );
+}
+
+interface AccountListItemProps {
+  account: Account;
+  wallet: WalletStore;
+  tokenId: string;
+  onPress: () => void;
+}
+
+export function AccountListItem({ account, wallet, tokenId, onPress }: AccountListItemProps) {
+  const tokenBalance = useGetAccountTokenBalance({ tokenId, account });
+  const availableBalance = tokenBalance?.availableBalance;
+  const quoteBalance = tokenBalance?.quoteBalance;
+
+  return (
+    <AccountListItemComponent
+      px="0"
+      accountName={account.name}
+      walletName={
+        <Text variant="caption01" lineHeight={16}>
+          {/* Should perhaps refactor account to have a wallet name?  Avoids using the wallet store here */}
```

Comment:
We need to use `WalletLoader` to get the `wallet` name. We could probably add that to the account object